### PR TITLE
YANG and text model updates for WDM tunnel model

### PR DESCRIPTION
**Pull request by aguoietf**
YANG and text update for the following issue: 

- fix typo in text
- added preferred-rx-channel-power as a transceiver constraint; 
- sync'ed with l0-types for rw/ro split of transceiver parameters; 
- added augments for reporting estimated transceiver properties on computed paths; 
- added augments for reporting actual transceiver properties on LSPs"

### CODE REVIEW

File: ietf-wdm-tunnel.tree [hunk 1]
```diff
@@ -3,69 +3,142 @@ module: ietf-wdm-tunnel
...
-       +--rw delta-power?                   l0-types:ratio-in-db
+       +--rw delta-power?                   l0-types:power-ratio
   augment /te:te/te:tunnels/te:tunnel/te:primary-paths
-            /te:primary-path/te:explicit-route-objects-always
+            /te:primary-path/te:explicit-route-objects
             /te:route-object-exclude-always/te:type
             /te:numbered-node-hop/te:numbered-node-hop:
     +--rw path-in-transceiver
-    |  +--rw transponder-id?        uint32
+    |  +--rw transponder-id?   uint32
```
Comment (italobusi) [inline ietf-wdm-tunnel.tree:27]: 1. I am wondering whether we need a new hop type for the 3R function where to put this information
2. should the container be named 'path-in-transponder' and 'path-out-transponder'?
3. how to provide constraints on the transponders at the ingress/egress of the tunnel?

File: ietf-wdm-tunnel.yang [hunk 2]
```diff
@@ -78,17 +76,17 @@ module ietf-wdm-tunnel {
   identity lower-first-wavelength-assignment {
     base l0-types:wavelength-assignment;
     description
-      "Assign first available wavelength starting from the
-       lowest frequency towards the highest frequency in
-       the allowable frequency range.";
+      "Allocate wavelengths in ascending order, beginning from the
+       lowest frequency and progressing toward the highest frequency
+       within the permissible frequency range.";
```
Comment (italobusi) [inline ietf-wdm-tunnel.yang:81]: 1. Should this identity be move to L0 types?

   See: https://github.com/ietf-ccamp-wg/ietf-ccamp-layer0-types-ext-RFC9093-bis/issues/92

2. Is there a reference where this wavelength assignment is defined?

File: ietf-wdm-tunnel.yang [hunk 3]
```diff
@@ -78,17 +76,17 @@ module ietf-wdm-tunnel {
...
 
   identity upper-first-wavelength-assignment {
     base l0-types:wavelength-assignment;
     description
-      "Assign first available wavelength starting from the
-       highest frequency towards the lowest frequency in
-       the allowable frequency range.";
+      "Allocate wavelengths in decending order, beginning from the
+       highest frequency and progressing toward the lowest frequency
+       within the permissible frequency range.";
```
Comment (italobusi) [inline ietf-wdm-tunnel.yang:89]: 1. Should this identity be move to L0 types?

   See: https://github.com/ietf-ccamp-wg/ietf-ccamp-layer0-types-ext-RFC9093-bis/issues/92

2. Is there a reference where this wavelength assignment is defined?

File: ietf-wdm-tunnel.yang [hunk 4]
```diff
@@ -121,12 +119,19 @@ module ietf-wdm-tunnel {
...
+	
+    leaf preferred-rx-channel-power {
+      type l0-types:power-dbm;
+      description
+        "The preferred channel received power.";
+    }
   }
 
-  grouping path-transceiver-config-grp {
+  grouping transceiver-config-grp {
```
Comment (italobusi) [inline ietf-wdm-tunnel.yang:131]: General: we do not need the '-grp' suffix for groupings

File: ietf-wdm-tunnel.tree [hunk 5]
```diff
@@ -3,69 +3,142 @@ module: ietf-wdm-tunnel
...
+       |  +--rw operational-modes*            string
        |  +--rw tx-tune-constraints
-       |     +--rw min-central-frequency?    frequency-thz
-       |     +--rw max-central-frequency?    frequency-thz
-       |     +--rw transceiver-tunability?   frequency-ghz
-       +--rw gsnr-margin?                   snr
+       |  |  +--rw min-central-frequency?    frequency-thz
+       |  |  +--rw max-central-frequency?    frequency-thz
+       |  |  +--rw transceiver-tunability?   frequency-ghz
+       |  +--rw line-coding-bitrate?          identityref
```
[the remark this answers is not in the document]
Comment (aguoietf) [reply ietf-wdm-tunnel.tree:11]: change line-coding-bitrate as a list (optional parameters)

[the remark this answers is not in the document]
Comment (aguoietf) [reply ietf-wdm-tunnel.tree:11]: add to text description the use of the WDM constraints (for limiting the possible choices when present)

File: ietf-wdm-tunnel.tree (hunk 1 above)
Comment (aguoietf) [reply ietf-wdm-tunnel.tree:27]: Looking into an example to check if this (path-in/path-out) transponder constructs stands to describe the ERO and RROs for WDM tunnel

File: ietf-wdm-tunnel.yang (hunk 2 above)
Comment (aguoietf) [reply ietf-wdm-tunnel.yang:81]: issue resolved by https://github.com/ietf-ccamp-wg/ietf-ccamp-layer0-types-ext-RFC9093-bis/pull/96

File: ietf-wdm-tunnel.yang (hunk 3 above)
Comment (aguoietf) [reply ietf-wdm-tunnel.yang:89]: issue resolved by https://github.com/ietf-ccamp-wg/ietf-ccamp-layer0-types-ext-RFC9093-bis/pull/96.
Need to remove the identityref here

File: ietf-wdm-tunnel.tree (hunk 5 above)
[the remark this answers is not in the document]
Comment (aguoietf) [reply ietf-wdm-tunnel.tree:11]: line-coding-bitrate is defined in l0-types in the grouping "common-transceiver-configured-param".
Q1: is line-coding-bitrate(s) per operational mode? If so it needs to be moved to under a container together with the operational mode (and under a list).
Q2 : "common-transceiver-configured-param" also includes tx-channel-power. If the tx-channel-power is not per operational mode then this grouping is not usable and we have to define separate line-coding-bitrate list instead.

[the remark this answers is not in the document]
Comment (sergiobelotti) [reply ietf-wdm-tunnel.tree:11]: Q1: operational and organizational modes have already their own line-coding-bitrate 
`grouping common-standard-organizational-mode {
    description
      "Common attributes used by standard and organizational 
      transceiver's modes.";
    
    leaf-list line-coding-bitrate {
      type identityref {
        base line-coding;
      }
      config false;
      description
        "The list of the bit rate/line coding of the optical 
        tributary signal supported by the transceiver.

        Reporting this list is optional when the standard or 
        organization mode supports only one bit rate/line coding.";
      reference 
        "ITU-T G.698.2 section 7.1.2";
    }
  } // grouping common-standard-organizational-mode`

Q2: tx-channel-power is also per operational- mode so is correctly in common for any modes

[the remark this answers is not in the document]
Comment (ggrammel) [reply ietf-wdm-tunnel.tree:11]: Additional comment:
Except for operational modes, Transceiver constraints are related to the tunnel selection and are meant to further constrain the resources available to the tunnel. All constraints are inclusive i.e. with an AND logic. E.g. a specific frequency range would need to select any of the operational modes AND constrain the tunnel to match only that range.

**Pull request merged**